NFT related Jira items

DennisDawson opened the pull request:
Fixes per TOK-52, RIPD-1799, RIPD-1800.

DennisDawson committed:
Add clarification for NFTokenOfferID response

Per TOK-52, providing a value with a valid format can result in a success response, even though no NFTokenOffer has been removed.

DennisDawson committed:
remove reference to lsfIssuerCanCancelOffers

This feature has been removed.

DennisDawson committed:
Add owner field description

Per https://ripplelabs.atlassian.net/browse/RIPD-1800, when the issuer burns a NFToken with tfBurnable set the burn request must include the Owner field.

DennisDawson committed:
Update TransferFee per spec change

Per https://ripplelabs.atlassian.net/browse/RIPD-1799, the spec changed to allow for a range of 0-50,000.

mDuo13 commented on content/references/protocol-reference/transactions/transaction-types/nftokencanceloffer.md:
```diff
@@ -41,8 +40,11 @@ This transaction removes the listed `NFTokenOffer` object from the ledger, if pr
-The transaction can succeed even if one or more of the IDs in the `TokenOffers` field do not refer to objects that currently exist in the ledger. (For example, those token offers might already have been taken.) The transaction fails with an error if one of the IDs points to an object that does exist, but is not a [NFTokenOffer](nftokenoffer.html) object.
+The transaction can succeed even if one or more of the IDs in the `TokenOffers` field do not refer to objects that currently exist in the ledger. (For example, those token offers might already have been taken.)
 
+It is important to note that if you inadvertently provide a `NFTokenID` rather than a `NFTokenOfferID`, you might receive a `tesSuccess` response. The reason is that when passed a properly formatted VECTOR256 value that is not found, the system assumes that the NFTokenOffer has already been deleted.
```
`NFTokenID` and `NFTokenOfferID` are not official names for data types, so it's confusing to put those in monospace font. But it works if you refer to the "ID" of those types of objects. Also, `tesSUCCESS` is the proper capitalization of the error code in the API. Also, the list is a "VECTOR256" but each item _in_ the list is a "Hash256" type (a raw 256-bit value of any kind—a slight misnomer since they were originally or mostly intended to hold hash values but in some cases hold other types of data). 

```suggestion
It is important to note that if you inadvertently provide the ID of a `NFToken` object rather than a `NFTokenOffer` object, you might receive a `tesSUCCESS` response. The reason is that when passed a properly formatted ID value that is not found, the system assumes that the `NFTokenOffer` has already been deleted.
```

DennisDawson replied:
This is fine, but need to delete the "a" in "...inadvertently provide *a* the ID of a...."

mDuo13 replied:
Edited my suggestion to fix that.

mDuo13 reviewed (changes requested):
Just a couple of nitpicks with the ID formatting stuff, otherwise looks good.

(I didn't know that the TransferFee format had changed though)

DennisDawson committed:
Update variable names in TokenOffers note.

Renaming the variable to nft_offer_index makes the distinction less critical. Ultimately we'll probably remove the note on line 45 entirely.

DennisDawson committed:
Merge branch 'master' into dd-NFT-Jira-Fixes20200322

amarantha-k reviewed (approved):
LGTM

manojsdoshi reviewed (approved):
LGTM

mDuo13 commented on content/references/protocol-reference/transactions/transaction-types/nftokencanceloffer.md:
```diff
@@ -40,8 +40,11 @@ This transaction removes the listed `NFTokenOffer` object from the ledger, if pr
-The transaction can succeed even if one or more of the IDs in the `NFTokenOffers` field do not refer to objects that currently exist in the ledger. (For example, those token offers might already have been taken.) The transaction fails with an error if one of the IDs points to an object that does exist, but is not a [NFTokenOffer](nftokenoffer.html) object.
+The transaction can succeed even if one or more of the IDs in the `NFTokenOffers` field do not refer to objects that currently exist in the ledger. (For example, those token offers might already have been deleted.) The transaction fails with an error if one of the IDs points to an object that does exist, but is not a [NFTokenOffer](nftokenoffer.html) object.
 
+It is important to note that if you inadvertently provide a `nft_id` rather than a `nft_offer_index`, you might receive a `tesSuccess` response. The reason is that when passed a properly formatted ID value that is not found, the system assumes that the `NFTokenOffer` has already been deleted.
```
Please do correct the casing of the error code. Otherwise this looks good.

```suggestion
It is important to note that if you inadvertently provide a `nft_id` rather than a `nft_offer_index`, you might receive a `tesSUCCESS` response. The reason is that when passed a properly formatted ID value that is not found, the system assumes that the `NFTokenOffer` has already been deleted.
```

DennisDawson committed:
Merge branch 'master' into dd-NFT-Jira-Fixes20200322
